src/Microsoft.DocAsCode.Common/FileAbstractLayer/LinkFileWriter.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Common
+{
+    using System.Collections.Generic;
+    using System.IO;
+
+    public class LinkFileWriter : FileWriterBase
+    {
+        private readonly Dictionary<RelativePath, PathMapping> _mapping =
+            new Dictionary<RelativePath, PathMapping>();
+
+        public LinkFileWriter(string outputFolder)
+            : base(outputFolder) { }
+
+        #region Overrides
+
+        public override void Copy(PathMapping sourceFilePath, RelativePath destFilePath)
+        {
+            var key = destFilePath.GetPathFromWorkingFolder();
+            _mapping[key] = new PathMapping(key, sourceFilePath.PhysicalPath)
+            {
+                Properties = sourceFilePath.Properties,
+            };
+        }
+
+        public override FileStream Create(RelativePath file)
+        {
+            var key = file.GetPathFromWorkingFolder();
+            PathMapping pm;
+            if (_mapping.TryGetValue(key, out pm) &&
+                pm.PhysicalPath.StartsWith(OutputFolder))
+            {
+                return File.Create(pm.PhysicalPath);
```

Comment:
no, it will truncate the content of file.

---
In reply to: [90165552](https://github.com/dotnet/docfx/pull/926#discussion_r90165552) [](ancestors = 90165552)